protocols/primary-backup/src/main/java/io/atomix/protocols/backup/roles/AsynchronousReplicator.java

```diff
@@ -68,6 +77,7 @@ public void close() {
     private final Queue<BackupOperation> operations = new LinkedList<>();
     private final NodeId nodeId;
     private final Scheduled backupTimer;
+    private CompletableFuture<BackupResponse> lastBackup = CompletableFuture.completedFuture(null);
```

Comment:
are add and maybeBackup invoked from the same thread? do we need to sync on backup? volatile?

Reply:
I was thinking something like this https://gist.github.com/johnou/55bf838142b38695830f7941368ce9fb ping @kuujo

Reply:
both of them invoked in the same threadContext, maybe I think it doesn't need synchronized